### ISSUE TITLE
500 error issue when opening filemanager window

### ISSUE THREAD

**Issue by Silent2000** [UniSharp/laravel-filemanager#1058]
Please check the following instructions before submitting a bug :
 * Make sure you are using the latest version.
 * Make sure you read [installation](http://unisharp.github.io/laravel-filemanager/installation), [integration](http://unisharp.github.io/laravel-filemanager/integration), and [upgrade](http://unisharp.github.io/laravel-filemanager/upgrade) document.

And provide the followings :
 * Operating system : CentOS 7
 * Laravel version : 7.28.2
 * Package version : 2.2
 * Steps to reproduce your issue : 
 * Screenshots of browser console : https://ibb.co/2vJbMkD

Hi. We have LFM, and S3 setup in filesystem.php as a disk. It works for six months perfectly, but today it's suddenly give us error 500, when opening window with filemanager to upload new files.

**Comment by majestica** [comment]
we have the same problem, this is due to the number of files in s3, when there are few of them, it seems normal when, for example, there are 10k+ images such problems, this is probably due to the method of obtaining a listing of objects

**Comment by Silent2000** [comment]
> we have the same problem, this is due to the number of files in s3, when there are few of them, it seems normal when, for example, there are 10k+ images such problems, this is probably due to the method of obtaining a listing of objects

Yes, support of s3 storage said that filemanager send request for each file when listing them. Is it possible to optimize or fix this?

**Comment by aspross** [comment]
Does anyone have a fix for this?

**Comment by majestica** [comment]
> Does anyone have a fix for this?

we wrote our own file manager with our own pagination, we could not work with this filemanager (we have more than 100k images at s3  and it just hung terribly)
